### PR TITLE
Remove v1-specific copy from ProjectDetailsForm

### PR DESCRIPTION
**Pull request by aeolianeth**
## What does this PR do and why?

This PR picks some of the changes from https://github.com/jbx-protocol/juice-interface/pull/551.

It makes `src/components/shared/forms/ProjectDetailsForm.tsx` form-only and includes no copy. This is so that we can re-use this form in V2.

## Screenshots or screen recordings

UI is unchanged

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

### CODE REVIEW

File: src/components/shared/forms/ProjectDetailsForm.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  infoUri: string
+  handle: string
+  logoUri: string
+  twitter: string
+  discord: string
+  payButton: string
+  payDisclosure: string
+}
+
+export default function ProjectDetailsForm({
```
Comment (aeolianeth) [inline src/components/shared/forms/ProjectDetailsForm.tsx:20]: This component is mostly the same is previously. I'll point out the specific changes below.

File: src/components/shared/forms/ProjectDetailsForm.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  form,
+  onFinish,
+  hideProjectHandle = false,
+}: {
+  form: FormInstance<ProjectDetailsFormFields>
+  onFinish: (values: ProjectDetailsFormFields) => void
```
Comment (aeolianeth) [inline src/components/shared/forms/ProjectDetailsForm.tsx:26]: `onSave` changed to `onFinished` to match Antd `<Form />` prop name

File: src/components/shared/forms/ProjectDetailsForm.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        formItemProps={{
+          rules: [{ required: true }],
+        }}
+        onChange={name => {
+          const val = name ? normalizeHandle(name) : ''
+          // Use `handle` state to enable ProjectHandle to validate while typing
+          form.setFieldsValue({ handle: val })
+        }}
+      />
+      {!hideProjectHandle && (
```
Comment (aeolianeth) [inline src/components/shared/forms/ProjectDetailsForm.tsx:42]: V2 doesn't use `handle`'s. This gives us the ability to hide the handle input for V2.

File: src/components/shared/forms/ProjectDetailsForm.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        initialUrl={form.getFieldValue('logoUri')}
+        onSuccess={logoUri => {
+          const prevUrl = form.getFieldValue('logoUri')
+          // Unpin previous file
+          form.setFieldsValue({ logoUri })
+          if (prevUrl) unpinIpfsFileByCid(cidFromUrl(prevUrl))
+        }}
+      />
+      <Form.Item>
+        <Button htmlType="submit" type="primary">
```
Comment (aeolianeth) [inline src/components/shared/forms/ProjectDetailsForm.tsx:71]: Remove `onClick` handler, rely on `Form`'s `onFinish` handler instead.

File: src/components/v1/V1Create/index.tsx [hunk 5]
```diff
@@ -333,10 +336,11 @@ export default function V1Create() {
                 color: viewed
                   ? colors.text.primary
                   : colors.text.action.primary,
-                border: viewed
-                  ? '1px solid ' + colors.stroke.secondary
-                  : '1px solid' + colors.stroke.action.primary,
-                borderLeftWidth: active ? 10 : 1,
+                borderColor: viewed
```
Comment (aeolianeth) [inline src/components/v1/V1Create/index.tsx:339]: Passing fix. This fixes a console error where apparently React doesn't like it when styles are overridden. In this case `borderLeftWidth` was overriding the above `border` style.

File: src/components/v1/V1Create/index.tsx [hunk 6]
```diff
@@ -578,15 +582,25 @@ export default function V1Create() {
...
+            </h1>
+            <p>
+              <Trans>
+                Changes to these attributes can be made at any time and will be
+                applied to your project immediately.
+              </Trans>
+            </p>
+            <ProjectDetailsForm
+              form={projectForm}
+              onFinish={() => {
```
Comment (aeolianeth) [inline src/components/v1/V1Create/index.tsx:597]: Note that `await projectForm.validateFields()` has been removed. Antd already calls this function before its `onFinish` callback is called, so calling `validateFields` again here would be redundant.

Comment (peripheralist) [reply src/components/v1/V1Create/index.tsx:597]: beautiful! being able to remove click handlers and externally handled validation feels like a big win. great call

**Comment by peripheralist** [comment]
Great PR, nicely scoped. great optimizations using the `onFinish` callback, and love reducing this component to not include any copy 👏

**Comment by aeolianeth** [comment]
Thanks for the review @peripheralist ! I'll assume your approval and merge ✅

**Pull request merged**